lib/matplotlib/table.py

```diff
@@ -178,50 +201,41 @@ def set_text_props(self, **kwargs):
         self._text.update(kwargs)
         self.stale = True
 
-
-class CustomCell(Cell):
-    """
-    A `.Cell` subclass with configurable edge visibility.
-    """
-
-    _edges = 'BRTL'
-    _edge_aliases = {'open':         '',
-                     'closed':       _edges,  # default
-                     'horizontal':   'BT',
-                     'vertical':     'RL'
-                     }
-
-    def __init__(self, *args, visible_edges, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.visible_edges = visible_edges
-
     @property
     def visible_edges(self):
         """
         The cell edges to be drawn with a line.
 
         Reading this property returns a substring of 'BRTL' (bottom, right,
-        top, left').
+        top, left).
 
         When setting this property, you can use a substring of 'BRTL' or one
-        of {'open', 'closed', 'horizontal', 'vertical'}.
+        of {'open', 'closed', 'horizontal', 'vertical', 'all', 'none'}
         """
         return self._visible_edges
 
     @visible_edges.setter
     def visible_edges(self, value):
+
         if value is None:
             self._visible_edges = self._edges
-        elif value in self._edge_aliases:
-            self._visible_edges = self._edge_aliases[value]
         else:
-            if any(edge not in self._edges for edge in value):
-                raise ValueError('Invalid edge param {}, must only be one of '
-                                 '{} or string of {}'.format(
-                                     value,
-                                     ", ".join(self._edge_aliases),
-                                     ", ".join(self._edges)))
-            self._visible_edges = value
+            lookup = dict(bottom='B', right='R', top='T', left='L')
+            value = ''.join((lookup.setdefault(x, x) for x in value))
```

Comment:
I don't understand what is supposed to happen here; since `value` is supposed to be a single string, this loop will go through every character. Thus you'd never hit any of the initially-set dictionary keys, and the result is just `value` all over again.

Thus, these two lines do nothing.

Reply:
See @timhoffm comment above suggesting the code allow the value to be a list of the longer more descriptive strings.

So this code allows for an iteration of strings, or just a string.   This was an attempt at adding that flexibility.

I'm not convinced it is worth the trouble, as this fix still needs some work since it would happily digest things like that "visible_edges=['clo', 'sed']"

Reply:
OK, so the docstring hasn't yet been updated to mention that, then?